lib/web3-contracts.js

```diff
@@ -61,6 +42,36 @@ export function useTokenDecimals(symbol) {
   return tokenDecimals.get(symbol)
 }
 
+export function useConnectorWeight() {
+  const marketMakerContract = useKnownContract('MARKET_MAKER')
+  const [address] = getKnownContract('COLLATERAL_TOKEN')
+  
+  const [reserveRatio, setReserveRatio] = useState()
+  const [collateralToken, setCollateralToken] = useState()
+
+  const fetchCollateral = useCallback(async () => {
+    try {
+      if(!marketMakerContract) {
+        return null
```

Comment:
I'm curious, throwing an error here would break the interface, right? Since we don't have any try..catch for it in the `useBondingCurvePrice` function. So I guess this line is fine as it is.

Reply:
yes! it does throw an error on start. I didnt use the same method as `useAllowance` for an ex, bc this function is called before we actually have a marketMakerContract (its called before the user connects to metamask).